R/functions__lazy.R

```diff
@@ -132,426 +142,283 @@ pl_implode = function(name) { # -> Expr
-#' - String: syntactic sugar for `pl$col(column)$approx_n_unique()`, returns Expr
-#' - Expr: syntactic sugar for `column$approx_n_unique()`, returns Expr
+#' Approximate count of unique values
 #'
-#' @keywords Expr_new
-#'
-#' @return Expr
+#' This function is syntactic sugar for `pl$col(...)$approx_n_unique()`,
+#' and uses the HyperLogLog++ algorithm for cardinality estimation.
+#' @inheritParams pl_head
+#' @inherit pl_head return
+#' @seealso
+#' - [`<Expr>$approx_n_unique()`][Expr_approx_n_unique]
+#' @examples
+#' df = pl$DataFrame(
+#'   a = c(1, 8, 1),
+#'   b = c(4, 5, 2),
+#'   c = c("foo", "bar", "foo")
+#' )
 #'
-#' @details The approx_n_unique is likely only warranted for large columns. See example.
-#' It appears approx_n_unique scales better than n_unique, such that the relative performance
-#' difference increases with column size.
+#' df$select(pl$approx_n_unique("a"))
 #'
-#' @examples
-#' # column as Series
-#' pl$approx_n_unique(pl$lit(1:4)) == 4
-#'
-#' # column as String
-#' expr = pl$approx_n_unique("bob")
-#' print(expr)
-#' pl$DataFrame(bob = 1:80)$select(expr)
-#'
-#' # colum as Expr
-#' pl$DataFrame(bob = 1:4)$select(pl$approx_n_unique(pl$col("bob")))
-#'
-#' # comparison with n_unique for 2 million integers. (try change example to 20 million ints)
-#' lit_series = pl$lit(c(1:1E6, 1E6:1, 1:1E6))
-#' system.time(pl$approx_n_unique(lit_series)$to_series()$print())
-#' system.time(pl$n_unique(lit_series)$to_series()$print())
-pl_approx_n_unique = function(column) { #-> int or Expr
-  pcase(
-    inherits(column, "RPolarsExpr"), result(column$approx_n_unique()),
-    is_string(column), result(pl$col(column)$approx_n_unique()),
-    or_else = Err(paste("arg [column] is neither Expr or String, but", str_string(column)))
-  ) |>
+#' df$select(pl$approx_n_unique("b", "c"))
+pl_approx_n_unique = function(...) {
+  result(pl$col(...)$approx_n_unique()) |>
     unwrap("in pl$approx_n_unique():")
 }
 
-#' Compute sum in one or several columns
+#' Sum all values.
 #'
-#' This is syntactic sugar for `pl$col(...)$sum()`.
-#'
-#' @param ...  One or several elements. Each element can be:
-#'  - Series or Expr
-#'  - string, that is parsed as columns
-#'  - numeric, that is parsed as literal
-#'
-#' @return Expr
-#' @keywords Expr_new
+#' Syntactic sugar for `pl$col(...)$sum()`.
+#' @inheritParams pl_head
+#' @inherit pl_head return
+#' @seealso
+#' - [`<Expr>$sum()`][Expr_sum]
+#' - [`pl$sum_horizontal()`][pl_sum_horizontal]
 #' @examples
-#' # column as string
-#' pl$DataFrame(iris)$select(pl$sum("Petal.Width"))
+#' df = pl$DataFrame(a = 1:2, b = 3:4, c = 5:6)
 #'
-#' # column as Expr (prefer pl$col("Petal.Width")$sum())
-#' pl$DataFrame(iris)$select(pl$sum(pl$col("Petal.Width")))
+#' df$select(pl$sum("a"))
 #'
-#' df = pl$DataFrame(a = 1:2, b = 3:4, c = 5:6)
+#' # Sum multiple columns
+#' df$select(pl$sum("a", "c"))
 #'
-#' # Compute sum in several columns
-#' df$with_columns(pl$sum("*"))
+#' df$select(pl$sum("^.*[bc]$"))
```

Comment:
It would be nice to have better column names to showcase regular expressions. I know this is the examples in the py-polars docs but not all examples there are great and they are refactoring them a lot